docs/engine/formats/dat.md

```diff
@@ -0,0 +1,161 @@
+# ZenGin DAT
+
+!!! abstract inline end "Quick Infos"
+    **Type:** Script Format<br/>
+    **Format Name:** DAT<br/>
+    **File Extension:** `.DAT`<br/>
+    **Class Name:** `zCParser`<br/>
+    **Encoding:** [Binary](../encodings/binary.md)<br/>
+
+*ZenGin* DAT files contain symbols and [bytecode](./bytecode.md) used by the VM of the engine.
+DAT files are the result of Daedalus script compilation process.
+
+## Format Description
+
+Compiled scripts are stored in a [binary](../encodings/binary.md) file which contains the following data. Also refer to the
+[Datatype Reference](../datatypes.md) for general information about often used datatypes.
+
+```c title="DAT Structure"
+struct DatFile {
+    byte version;
+
+    uint symbolCount;
+    uint symbolIds[/* symbolCount */]; // (1)
+    zCPar_Symbol symbolTable[/* symbolCount */];
+
+    uint bytecodeSize;
+    byte bytecode[/* bytecodeSize */]; // (2)
+};
+```
+
+1. Symbol IDs sorted lexicographically
+2. Read [bytecode](./bytecode.md) page for more info about this data.
+
+### Symbol
+
+```c title=""
+struct zCPar_Symbol {
+    byte isNamed; // (1) 
+    SymbolName name;
+
+    SymbolParameters params;
+    SymbolCodeSpan span;
+    SymbolData data;
+    int parent; // (2)
+};
+```
+
+1. Does this symbol have a name
+2. Parent type, `-1` means `None`
+
+#### Symbol Name
+
+```c title="If isNamed == 1"
+struct SymbolName {
+    string name;
+};
+```
+```c title="If isNamed == 0"
+struct SymbolName {};
+```
```

Comment:
I think this can be condensed into a note.

Reply:
Changed to a content tab to make it consistent with other conditional data layouts.